### PR TITLE
feat: automatically generate the credentialName in the Login Component

### PR DESCRIPTION
**Pull request by karanngi**
Hello Team,
This PR address the issue to automatically generate the credentialName in the Login Component.

Closes this  #37

### CODE REVIEW

File: cmd/harbor/root/login.go [hunk 1]
```diff
@@ -32,15 +32,19 @@ func LoginCommand() *cobra.Command {
...
-				Password: Password,
-				Name:     Name,
+				Username: username,
+				Password: password,
+				Name:     name,
+			}
+
+			if loginView.Name == "" && loginView.Server != "" && loginView.Username != "" {
+				loginView.Name = fmt.Sprintf("%s@%s", loginView.Username, utils.SanitizeServerAddress(loginView.Server))
+
```
Comment (bupd) [inline cmd/harbor/root/login.go:42]: this is redundant, please remove.
```suggestion
```

File: pkg/views/login/create.go [hunk 2]
```diff
@@ -49,9 +53,18 @@ func CreateView(loginView *LoginView) {
...
+				PlaceholderFunc(func() string {
+					return fmt.Sprintf("%s@%s", loginView.Username, utils.SanitizeServerAddress(loginView.Server))
+				}, &loginView).
+				SuggestionsFunc(func() []string {
+					return []string{
+						fmt.Sprintf("%s@%s", loginView.Username, utils.SanitizeServerAddress(loginView.Server)),
+					}
+				}, &loginView).
 				Validate(func(str string) error {
 					if str == "" {
```
Comment (bupd) [inline pkg/views/login/create.go:66]: Update `Name of Credential` `validate block` with this code 

```suggestion
				Validate(func(str string) error {
					if str == "" {
						loginView.Name = fmt.Sprintf("%s@%s", loginView.Username, utils.SanitizeServerAddress(loginView.Server))
						return nil
					}
					return nil
				}),

```

This ensures that when `Credential name` is empty it is autofilled with the default value.

**Review by bupd — CHANGES_REQUESTED** [review]

**Comment by Vad1mo** [comment]
related https://github.com/goharbor/harbor-cli/pull/148

**Review by bupd — CHANGES_REQUESTED** [review]
please add the necessary changes and fix merge conflict.

**Comment by bupd** [comment]
https://github.com/goharbor/harbor-cli/pull/147#discussion_r1716251981

Committing these suggestions will make this PR ready for merge

**Comment by bupd** [comment]
closing in favor of (#250)

**Pull request closed without merging**